lib/postrank-uri.rb

```diff
@@ -142,8 +143,24 @@ def unescape(uri)
       str
     end
 
+    def unescape_unreserved(uri)
+      u = parse(uri)
+      u.query = u.query.tr('+', ' ') if u.query
+      str = u.to_s.gsub(URIREGEX[:unescape]) do
+        code = $1
+        next code if code =~ URIREGEX[:reserved]
```

Comment:
Matching against URIREGEX[:reserved] could be replaced by checking if code is in a set of reserved strings. Not sure if that's faster than a regex, but I suspect it might be.